### DISCUSSION TITLE
User root can't auth via pamltsp

### DISCUSSION THREAD

**Discussion by brcisna** [ltsp/ltsp#248] [Q&A]
Hello All,

When trying to run virt-machine-manager on diskless workstation i get prompted for root password. This user is added to all the necessary groups to launch virt-manager without root password prompt. And can open virt-machine-manager on server without issue. 
Anyway, when i enter root password the above message is displayed.
This may be the correct behaivior for root access?

Have used LTSP for about 15 years but brand new to this gen of LTSP. Still getting it all figured out.

Thank You

**Comment by alkisg** [comment]
Hi brcisna,

This tracker is for bugs: https://github.com/ltsp/ltsp/issues
This tracker is for discussions: https://github.com/ltsp/community/issues
Your questions are discussions, so I moved the issues you filed to the community tracker.

Are you using chrootless, a chroot, or a VM? I haven't used virt manager so let's ask help from others too.

**Comment by brcisna** [comment]
Hi Alkus

This is a chrootless setup. Sorry I should have given better details.
Also in addition to the above message when sitting at a client machine i cannot do an su  ,, and get root?

Thank You

**Comment by alkisg** [comment]
LTSP doesn't read or transfer the server "shadow", for safety reasons. This means that the password hashes for both "user" and "root" do not exist in the client.

When a user authenticates, an SSH connection is made to the server, and the result of the SSH authentication is used in pamltsp.
But usually the "root" user isn't configured for SSH authentication.

So, the normal way is to use "sudo -i", which asks for the user password if the user is in the sudo group; not "su -", which asks for the root password.
If you do want to use root, you would need to set a password for the clients only, separate to the server root password. To see how to do that, see the example right before the last one in the ltsp.conf man page: https://ltsp.org/man/ltsp.conf/

```shell
# Set the root password to "qwer1234" for all clients.
...
```

**Comment by alkisg** [comment]
Another thing: if you install virt manager, and this creates a system group, then you need to run `ltsp initrd` and reboot your client.

The /etc/passwd and /etc/group files are contained in the ltsp.img additional initrd image, so whenever you change these files, you need to run `ltsp initrd`, otherwise the clients won't know about any groups that virt manager has created.

**Comment by brcisna** [comment]
Hi Alkis,

Thank You for these explanations. I am trying to figure out how i can
launch virt-manager on the client machine. I have re-run ltsp initrd and
still i get  root password needed on client machine, and of course fails.
This same user is added to the libvirt groups,,,of about three,items,and
can launch virt-manager on server no problems.

Thank You

**Comment by alkisg** [comment]
Hi brcisna,

as I don't have virt-manager, you'd either need to wait for feedback from someone else, or come to the IRC channel for live troubleshooting; to do that, go to ltsp.org and click the IRC link.
